src/app/applications/datatarget/datatarget-test-connection/datatarget-test-connection.component.ts

```diff
@@ -0,0 +1,129 @@
+  editorJsonOutpuOptions = {
+    theme: "vs",
+    language: "json",
+    autoIndent: true,
+    roundedSelection: true,
+    minimap: { enabled: false },
+    readOnly: true,
+  };
+
+  public iotDevices: IotDevice[];
+  public payloadDecoders: PayloadDecoder[];
+  public payloadData = "";
+  public testResponse = "";
+  public decodedData = "";
+  public dataTarget: Datatarget;
+  public payloadDecoderId: number;
+  public loading = false;
+  protected readonly DataTargetType = DataTargetType;
+  private applicationId: number;
+  private dataTargetId: number;
+  private iotDeviceId: number;
+  private subscriptions = [];
+
+  constructor(
+    private deviceService: IoTDeviceService,
+    private dataTargetService: DatatargetService,
+    private route: ActivatedRoute,
+    private payloadDecoderService: PayloadDecoderService,
+    private payloadDeviceDataTargetService: PayloadDeviceDatatargetService,
+    public applicationService: ApplicationService
+  ) {}
+
+  ngOnInit(): void {
+    this.applicationId = +this.route.parent.snapshot.paramMap.get("id");
+    this.dataTargetId = +this.route.parent.snapshot.paramMap.get("datatargetId");
+    this.getDevices();
+    this.getDatatarget();
+    this.getPayloadDecoders();
+  }
+
+  ngOnDestroy() {
+    this.subscriptions.forEach(s => s?.unsubscribe());
+  }
+
+  getDevices() {
+    const deviceSubscription = this.applicationService
+      .getApplication(this.applicationId)
+      .subscribe((application: Application) => {
+        this.iotDevices = application.iotDevices.sort((a, b) => a.name.localeCompare(b.name, "en", { numeric: true }));
+      });
+
+    this.subscriptions.push(deviceSubscription);
+  }
+
+  getLastPackage(event: MatSelectChange) {
+    this.iotDeviceId = event.value;
+    const deviceSub = this.deviceService.getIoTDevice(event.value).subscribe((device: IotDevice) => {
+      if (device.latestReceivedMessage) {
+        this.payloadData = JSON.stringify(device.latestReceivedMessage.rawData, null, 2);
+      }
+    });
+
+    const connectionSub = this.payloadDeviceDataTargetService.getByDataTarget(this.dataTargetId).subscribe(c => {
+      const connection = c.data.find(x => x.iotDevices.some(d => d.id === event.value));
+      this.payloadDecoderId = connection.payloadDecoder.id;
+    });
+
+    this.subscriptions.push(deviceSub, connectionSub);
+  }
+
+  pingDataTarget(dataPackage?: string) {
+    this.loading = true;
+    this.dataTargetService
+      .testDataTarget({
+        dataTargetId: this.dataTargetId,
+        iotDeviceId: this.iotDeviceId,
+        payloadDecoderId: this.payloadDecoderId,
+        dataPackage: dataPackage,
+      })
+      .subscribe(
+        response => {
+          this.testResponse = response?.result ? JSON.stringify(response.result, null, 2) : "";
+          this.decodedData = response?.decodedPayload ? JSON.stringify(response.decodedPayload, null, 2) : "";
+        },
+        error => {
+          console.log(error);
```

Comment:
loading skal også sættes til false her. Og så skal der måske laves en snackbar eller en anden form for besked på, at der skete en fejl.

Reply:
Den bliver sat til false, den nederste funktion er "onComplete" altså noget der altid sker lige meget om det er success eller error

Reply:
Den eneste der realistisk bør ske at hvis dekodningen fejler, men min assumption om featuren har været at denne del bruges når man har sin dekoder sat op og testet